pagopa/io-app
# chore(IT Wallet): [SIW-1927] Add pull-to-refresh to the wallet screen

## Short description
This PR enables the "pull to refresh" feature in the wallet home screen. 

## List of changes proposed in this pull request
- Fixed in issue in the `IOScrollView` component, which was causing the RefreshIndicator to ne be displayed correctly. See more here: https://github.com/facebook/react-native/issues/27324 (cc @dmnplb)
- Refactored `WalletHomeScreen` removing unnecessary extra components
- Moved analytics sagas from `watchWalletSaga` to dedicated `handleWalletAnalyticsSaga`
- Added `walletUpdate` action, handled by `handleWalletUpdateSaga`, which updates the wallet content
- Improved wallet placeholders handling in the `watchWalletSaga` handler and wallet reducer: placeholders are no more replacing existing cards with the same key
- Added `isWalletScreenRefreshingSelector` selector, to get the wallet refresh state

## How to test
Navigate to the wallet screen, try to "pull to refresh" and see if the content is being updated/refreshed and the refresh indicator works as expected.

## Demo


<video src="https://github.com/user-attachments/assets/910a2a56-8434-4006-b32b-d7995a0c18d2" />
